packages/yourloops/components/private-route.tsx

```diff
@@ -0,0 +1,35 @@
+// A wrapper for <Route> that redirects to the login
+
+import React from "react";
+import { Redirect, Route, RouteProps } from "react-router-dom";
+import { useAuth } from "../lib/auth/hook/use-auth";
+
+interface PrivateRouteProps extends RouteProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  component: any,
+}
+
+// screen if you're not yet authenticated.
+function PrivateRoute( props : PrivateRouteProps ) : JSX.Element {
+  const auth = useAuth();
+  const { component: Component, ...rest } = props;
+  return (
+    <Route
+      {...rest}
+      render={( routeProps ) =>
+        auth.isLoggedIn() ? (
+          <Component {...routeProps} />
+        ) : (
+            <Redirect
+              to={{
+                pathname: "/login",
+                state: { from: routeProps.location },
+              }}
+            />
+        )
+      }
+    />
+  );
+}

```

Comment:
A little bit too complicated code, could you slip it ?
Example:
```suggestion
/**
 * Perform a redirect if you are not authenticated yet
 * @param props RouteProps with the component to render.
 */
function PrivateRoute(props: PrivateRouteProps): JSX.Element {
  const auth = useAuth();
  const { component: Component, ...rest } = props;

  const render = (routeProps: RouteProps) => {
    if (auth.isLoggedIn()) {
      return <Component {...routeProps} />;
    }
    return <Redirect to={{ pathname: "/", state: { from: routeProps.location } }} />;
  };

  return <Route {...rest} render={render} />;
}
```